### PR TITLE
ref(hydration): Refactor side by side diff

### LINKED ISSUE #78120: [Hydration Errors] Refactor Side-by-side diff to use new /player/replayPlayer.tsx component(s)
There is a new `<ReplayPlayer>` component and we should use it within the side-by-side diff tool!

New one: https://github.com/getsentry/sentry/blob/master/static/app/components/replays/player/replayPlayer.tsx

The old one is: https://github.com/getsentry/sentry/blob/master/static/app/components/replays/replayPlayer.tsx

The big difference with the new one is that we're not using the old `<ReplayContext>` anymore, so all the hacks and stuff in there are going to be gone. On the flip side though, there are a few Providers that need to be put in place instead of that one single thing, it's not a big deal though!

The Slider diff is a good example for how the code will turn out: https://github.com/getsentry/sentry/blob/7b3f7bd38503664b33afa7aa05eabca098a2c66f/static/app/components/replays/diff/replaySliderDiff.tsx#L111-L136

Basically this task is to look at that link above ^, then apply it to here: https://github.com/getsentry/sentry/blob/7b3f7bd38503664b33afa7aa05eabca098a2c66f/static/app/components/replays/diff/replaySideBySideImageDiff.tsx#L31-L55

**Outcomes**

This should help with render issues, where (for example) one side of the diff is blank. Also, using this new set of components for replay overall will be a good thing imo, so we can refactor to get there step-by-step.

### PR DESCRIPTION
**Pull request by c298lee**
Refactors the side by side diff to use the new `<ReplayPlayer>` component 
Before:

https://github.com/user-attachments/assets/6b84f895-c9c2-4433-a80d-61384f747a75

After:

https://github.com/user-attachments/assets/eaec2962-fa0b-41cf-bbcf-90bdca282e0c

Closes https://github.com/getsentry/sentry/issues/78120

### CODE REVIEW

File: static/app/components/replays/diff/replaySideBySideImageDiff.tsx [hunk 1]
```diff
@@ -28,31 +29,24 @@ export function ReplaySideBySideImageDiff({leftOffsetMs, replay, rightOffsetMs}:
       </DiffHeader>
 
       <ReplayGrid>
-        <ReplayContextProvider
-          analyticsContext="replay_comparison_modal_left"
```
Comment (michellewzhang) [inline static/app/components/replays/diff/replaySideBySideImageDiff.tsx:32]: do we lose analytics tracking by removing this?

Comment (ryan953) [reply static/app/components/replays/diff/replaySideBySideImageDiff.tsx:32]: I see it's used for play/pause tracking: https://github.com/getsentry/sentry/blob/7b3f7bd38503664b33afa7aa05eabca098a2c66f/static/app/components/replays/replayContext.tsx#L494-L500

and also for fullscreen tracking. Neither are applicable to this hydration tool, so we should be good. But we'll need to include it in the new system as we refactor more.

File: static/app/components/replays/diff/replaySliderDiff.tsx
Comment (ryan953) [inline static/app/components/replays/diff/replaySliderDiff.tsx:1]: damn, it was `any` before :(

oops. I see this is having a good impact on replayPlayerEventsContext below...

File: static/app/utils/replays/playback/providers/replayPlayerEventsContext.tsx [hunk 2]
```diff
@@ -10,9 +10,11 @@ export function ReplayPlayerEventsContextProvider({
   replay,
 }: {
   children: React.ReactNode;
-  replay: ReplayReader;
+  replay: ReplayReader | null;
```
Comment (ryan953) [inline static/app/utils/replays/playback/providers/replayPlayerEventsContext.tsx:13]: it it really null sometimes, or is this to make the types happy?

Comment (ryan953) [reply static/app/utils/replays/playback/providers/replayPlayerEventsContext.tsx:13]: it's a pain, but might be possible to chase around the types and remove a bunch of `| null`

File: static/app/components/replays/diff/replaySideBySideImageDiff.tsx [hunk 3]
```diff
@@ -28,31 +29,24 @@ export function ReplaySideBySideImageDiff({leftOffsetMs, replay, rightOffsetMs}:
...
+              <ReplayPlayerStateContextProvider>
+                <ReplayPlayerMeasurer measure="width">
+                  {style => <ReplayPlayer style={style} offsetMs={leftOffsetMs} />}
+                </ReplayPlayerMeasurer>
+              </ReplayPlayerStateContextProvider>
+            </Border>
             <Border>
-              <ReplayPlayer isPreview />
+              <ReplayPlayerStateContextProvider>
+                <ReplayPlayerMeasurer measure="both">
```
Comment (ryan953) [inline static/app/components/replays/diff/replaySideBySideImageDiff.tsx:43]: above we've got `measure="width"`
idk which is best, i forgot what's what. but i expected them to be the same.

**Review by ryan953 — APPROVED** [review]

File: static/app/utils/replays/playback/providers/replayPlayerEventsContext.tsx (hunk 2 above)
Comment (c298lee) [reply static/app/utils/replays/playback/providers/replayPlayerEventsContext.tsx:13]: I'll do this in a follow up PR!

File: static/app/components/replays/diff/replaySideBySideImageDiff.tsx (hunk 3 above)
Comment (c298lee) [reply static/app/components/replays/diff/replaySideBySideImageDiff.tsx]: oops, it's fixed, should be `width`

**Pull request merged**